test/new-e2e/tests/agent-metric-logs/linux-log-agent/utf_test.go

```diff
@@ -0,0 +1,122 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package logagent
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	fi "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	e2e "github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
+)
+
+//go:embed log-config/utf-16-le-log-config.yaml
+var utfLittleEndianLogConfig []byte
+
+//go:embed log-config/utf-16-be-log-config.yaml
+var utfBigEndianLogConfig []byte
+
+var service = "utfservice"
+
+// UTF defines a test suite for the log agent interacting with a virtual machine and fake intake.
+type UtfSuite struct {
+	e2e.Suite[e2e.FakeIntakeEnv]
+	DevMode bool
+}
+
+// TestE2EVMFakeintakeSuite runs the E2E test suite for the log agent with a VM and fake intake.
+func TestUtfSuite(t *testing.T) {
+	s := &UtfSuite{}
+	_, s.DevMode = os.LookupEnv("TESTS_E2E_DEVMODE")
+	e2e.Run(t, s, e2e.FakeIntakeStackDef())
+}
+
+func (s *UtfSuite) generateUtfLog(endianness, content string) error {
+	utfLogGenerationCommand := fmt.Sprintf(`sudo python3 -c "f = open('/var/log/hello-world-utf.log', 'ab'); t = '%s\n'.encode('utf-16-%s'); f.write(t); f.close()"`, content, endianness)
```

Comment:
This is appending the contents to the log file in `binary` mode (`ab`). Two questions:
- Is `binary` required for this test?
- Is appending intentional vs overwriting the contents?

Reply:
1. `'b'` is needed due to `.encode`. The code breaks otherwise.
2. I am appending here since I am using the same `hello-world-utf.log` log file.

Reply:
1. Sounds good 👍 
2. So we don't care if there are already contents in the log file from a previous test, correct?

I just want to be intentional about appending/overwriting contents because as we add more tests this is all going to get more complex.

Reply:
We don't care in this case since old logs would be in different encodings but I'll change it to overwrite to avoid any potential issues in the future 👍.